sql/firefox_accounts/fxa_amplitude_email_clicks/view.sql

```diff
@@ -0,0 +1,220 @@
+    ) AS user_id,
+    email_providers_over_10k.email_provider,
+  FROM
+    customers_deduped
+  LEFT JOIN
+    email_providers_over_10k
+  USING
+    (email_provider)
+)
+SELECT
+  UNIX_MILLIS(clicks.EventDate) AS `time`,
+  customers.user_id,
+  ARRAY_TO_STRING([customers.user_id, clicks.SendId, string(clicks.EventDate)], '-') AS insert_id,
+  FORMAT(
+    '{%t}',
+    ARRAY_TO_STRING(
+      ARRAY(
+        SELECT
+          FORMAT('"%t":"%t"', key, value)
+        FROM
+          UNNEST(
+            [
+              STRUCT('email_provider' AS key, customers.email_provider AS value),
+              STRUCT(
+                'utm_source' AS key,
+                REGEXP_EXTRACT(clicks.URL, r"[?&]utm_source=([^&]+)") AS value
+              ),
+              STRUCT(
+                'utm_medium' AS key,
+                REGEXP_EXTRACT(clicks.URL, r"[?&]utm_medium=([^&]+)") AS value
+              ),
+              STRUCT(
+                'utm_campaign' AS key,
+                REGEXP_EXTRACT(clicks.URL, r"[?&]utm_campaign=([^&]+)") AS value
+              ),
+              STRUCT(
+                'utm_content' AS key,
+                REGEXP_EXTRACT(clicks.URL, r"[?&]utm_content=([^&]+)") AS value
+              ),
+              STRUCT(
+                'utm_term' AS key,
+                REGEXP_EXTRACT(clicks.URL, r"[?&]utm_term=([^&]+)") AS value
+              )
+            ]
+          )
+        WHERE
+          value IS NOT NULL
+      ),
+      ','
+    )
+  ) AS user_properties,
+  FORMAT(
+    '{%t}',
+    ARRAY_TO_STRING(
+      ARRAY(
+        SELECT
+          FORMAT('"%t":"%t"', key, value)
+        FROM
+          UNNEST(
+            [
+              STRUCT('service' AS key, customers.service AS value),
+              STRUCT('email_alias' AS key, clicks.alias AS value),
+              STRUCT('email_type' AS key, clicks.EventType AS value),
+              -- sendjobs.EmailName is parsed via regex; each regex differs only
+              -- in which group is doing the capturing (no ?: prefix);
+              -- see original regex in https://github.com/mozilla/fxa-amplitude-send/blob/89bfaef20a1d978fce2dccddc0155699a17ac172/marketing.js#L109
+              STRUCT(
+                'email_sender' AS key,
+                REGEXP_EXTRACT(
+                  sendjobs.EmailName,
+                  r"([A-Za-z]+)_(?:[A-Z]+)_[A-Z]*_[0-9]{4}_[A-Z]+_(?:[A-Z]+|DESK[_ ][A-Z]+)_(?:.+?)_(?:ALL|[A-Z]{2})_(?:[A-Z]{2,4})_(?:[A-Z-]+)(?:_[A-Za-z0-9]*)?"
+                ) AS value
+              ),
+              STRUCT(
+                'email_region' AS key,
+                REGEXP_EXTRACT(
+                  sendjobs.EmailName,
+                  r"(?:[A-Za-z]+)_([A-Z]+)_[A-Z]*_[0-9]{4}_[A-Z]+_(?:[A-Z]+|DESK[_ ][A-Z]+)_(?:.+?)_(?:ALL|[A-Z]{2})_(?:[A-Z]{2,4})_(?:[A-Z-]+)(?:_[A-Za-z0-9]*)?"
+                ) AS value
+              ),
+              STRUCT(
+                'email_focus' AS key,
```

Comment:
This one is new?

Reply:
Yes, I'm including it here for completeness, but it's a judgment call. I could be convinced to drop and keep it parallel with the pipeline it's replacing.